src/store/messages/saga.ts

```diff
@@ -277,16 +277,19 @@ export function* stopSyncChannels(action) {
 
 export function* receiveNewMessage(action) {
   const { channelId, message } = action.payload;
+  const channelIdWithoutPrefix = channelId.replace('sendbird_group_channel_', '');
 
-  const cachedMessageIds = [...(yield select(getCachedMessageIds(channelId)))];
-  const currentMessages = yield select(rawMessagesSelector(channelId));
+  const cachedMessageIds = [...(yield select(getCachedMessageIds(channelIdWithoutPrefix)))];
+  const currentMessages = yield select(rawMessagesSelector(channelIdWithoutPrefix));
 
   let messages = [];
+  let cachedMessageLength = 0;
 
   if (cachedMessageIds.length) {
     messages = [
       ...currentMessages,
     ];
+    cachedMessageLength = cachedMessageIds.length;
```

Comment:
I realize this isn't really part of your change but why do line 298 and line 300 return different types of things?
```rb
if (messageId === firstCachedMessageId) {
        cachedMessageIds.shift();
        return message;
      } else {
        return messageId;
      }
```
One is a message object and one is a messageId. That seems like it could be a source of bugs too.

Reply:
We can create a task for a refactor.